src/core_plugins/metrics/public/kbn_vis_types/editor_controller.js

```diff
@@ -53,7 +52,13 @@ function ReactEditorControllerProvider(Private, localStorage, config) {
           fetchFields(indexPatterns).then(fields => {
             this.vis.fields = { ...fields, ...this.vis.fields };
             const Component = this.vis.type.editorConfig.component;
-            render(<Component config={config} vis={this.vis} visData={visData} renderComplete={resolve}/>, this.el);
+            render(<Component
+              config={config}
+              vis={this.vis}
+              savedObj={savedObj}
+              timeRange={params.timeRange}
+              renderComplete={resolve}
```

Comment:
I would suggest we also pass in `inEditorMode={true}` here. That way we could exchange that `this.props.visData.type` in `vis_editor` by a more appropriate `!this.props.inEditorMode`, since that component is only used by this one here and directly in the vis infrastructure.